src/hotspot/share/prims/jvmtiEnvThreadState.cpp

```diff
@@ -397,12 +397,8 @@ void JvmtiEnvThreadState::reset_current_location(jvmtiEvent event_type, bool ena
       // so get current location with direct handshake.
       GetCurrentLocationClosure op;
       Thread *current = Thread::current();
-      if (thread->is_handshake_safe_for(current)) {
-        op.do_thread(thread);
-      } else {
-        Handshake::execute(&op, thread);
-        guarantee(op.completed(), "Handshake failed. Target thread is not alive?");
-      }
+      Handshake::execute(&op, thread);
+      guarantee(op.completed(), "Handshake failed. Target thread is not alive?");
```

Comment:
I much prefer that the current-thread case is internalised by `Handshake::execute` now. The code creating the handshake op shouldn't have to worry about current thread or not.